pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -281,7 +281,16 @@ func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec,
 }
 
 func (p *provider) GetCloudConfig(spec v1alpha1.MachineSpec) (config string, name string, err error) {
-	return "", "", nil
+	c, _, err := p.getConfig(spec.ProviderSpec)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to parse config: %v", err)
+	}
+	cc := kubevirttypes.CloudConfig{
+		Kubeconfig: c.Kubeconfig,
+	}
+	ccs, err := cc.String()
+
+	return ccs, "kubevirt", err
```

Comment:
Nit: you can use `CloudProviderKubeVirt` type, found in `pkg/providerconfig/types/types.go` instead of explicit typing "kubevirt"

Reply:
done